### PR TITLE
[ci] Parameterize pipeline and improve azure pipeline

### PR DESCRIPTION
**Pull request by liushilongbuaa**
1. Parameterize pipeline. When checkout new release branch, we don't need manually setup azp.
2. Use common lib pipeline instead of buildimage pipeline to get higher success rate.
3. Clean workspace when downloading external packages. This will ensure package version matching.

### CODE REVIEW

File: .azure-pipelines/build-docker-sonic-vs-template.yml [hunk 1]
```diff
@@ -32,35 +32,44 @@ jobs:
...
+      path: $(Build.ArtifactStagingDirectory)/download
+    displayName: "Download pre-stage built ${{ parameters.swss_artifact_name }}"
   - task: DownloadPipelineArtifact@2
     inputs:
       source: specific
       project: build
-      pipeline: 1
+      pipeline: Azure.sonic-buildimage.official.vs
       artifact: sonic-buildimage.vs
       runVersion: 'latestFromBranch'
```
Comment (xumia) [inline .azure-pipelines/build-docker-sonic-vs-template.yml:59]: If latestFromBranch is specified, we may need to specify the branchName, the default value is refs/heads/master.
See https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/download-build-artifacts?view=azure-devops

File: .azure-pipelines/test-docker-sonic-vs-template.yml [hunk 2]
```diff
@@ -18,29 +18,36 @@ jobs:
       ls -A1 | xargs -I{} sudo rm -rf {}
     displayName: "Clean workspace"
   - checkout: self
+    clean: true
+    displayName: "Checkout sonic-swss-common"
+  - checkout: sonic-swss
```
Comment (xumia) [inline .azure-pipelines/test-docker-sonic-vs-template.yml:23]: Maybe we can use the Azure template extends to set the correct branch name in the template.
Currently, the repository resource does not access a dynamic branch name, not quite sure if we can move into the template.

File: .azure-pipelines/build-docker-sonic-vs-template.yml (hunk 1 above)
Comment (liushilongbuaa) [reply .azure-pipelines/build-docker-sonic-vs-template.yml:59]: Default value is ok.
Although remove this line, pipeline still use master branch.

File: .azure-pipelines/test-docker-sonic-vs-template.yml (hunk 2 above)
Comment (liushilongbuaa) [reply .azure-pipelines/test-docker-sonic-vs-template.yml:23]: resources can't appear after steps. we need to use script to get correct branch name.
So, using extends can't help dynamic branch name in resource.

File: azure-pipelines.yml [hunk 3]
```diff
@@ -2,11 +2,26 @@
 # Build your C/C++ project with GCC using make.
 # Add steps that publish test results, save build artifacts, deploy, and more:
 # https://docs.microsoft.com/azure/devops/pipelines/apps/c-cpp/gcc
+pr:
+- master
+- 202???
+- 201???
```
Comment (xumia) [inline azure-pipelines.yml:8]: Can we trigger for all branches?

File: azure-pipelines.yml [hunk 4]
```diff
@@ -2,11 +2,26 @@
...
 
 trigger:
+- master
+- 202???
+- 201???
```
Comment (xumia) [inline azure-pipelines.yml:13]: Can we trigger the ci build for all branches, and add batch=true?

File: azure-pipelines.yml (hunk 3 above)
Comment (liushilongbuaa) [reply azure-pipelines.yml:8]: There are some branches other than release branch.
It seems make sence to add some filter.

File: azure-pipelines.yml (hunk 4 above)
Comment (liushilongbuaa) [reply azure-pipelines.yml]: Add batch

**Review by xumia — APPROVED** [review]

**Pull request merged**